plugins/woocommerce/client/blocks/assets/js/blocks/product-details/edit.tsx

```diff
@@ -51,8 +63,49 @@ const TEMPLATE: InnerBlockTemplate[] = [
 	],
 ];
 
-const Edit = () => {
-	return <InnerBlocks template={ TEMPLATE } />;
+/**
+ * Check if block is inside a Query Loop with non-product post type
+ *
+ * @param {string} clientId The block's client ID
+ * @param {string} postType The current post type
+ * @return {boolean} Whether the block is in an invalid Query Loop context
+ */
+const useIsInvalidQueryLoopContext = ( clientId: string, postType: string ) => {
+	return useSelect(
+		( select ) => {
+			const blockParents = select(
+				blockEditorStore
+			).getBlockParentsByBlockName( clientId, 'core/post-template' );
+			return blockParents.length > 0 && postType !== 'product';
+		},
+		[ clientId, postType ]
+	);
+};
```

Comment:
@imanish003 just a note here, given you are moving this into a re-usable function.
I changed the implementation a bit, to better make use of the `useSelect` function, this also avoid the use of `useMemo` as in the current implementation: https://github.com/woocommerce/woocommerce/blob/c75fa32e72214c9db5235cb82b51aec978444cea/plugins/woocommerce/client/blocks/assets/js/blocks/product-description/edit.tsx#L183-L190

I guess I overlooked this when reviewing your PR.

Reply:
Ah, good point. I'll make the changes accordingly. Thanks for pointing this out 🙇